### ISSUE TITLE
Limit number of concurrently processing nodes

### ISSUE THREAD

**Issue by lukors** [lukors/kanter_core#22]
This should be done after #28.

Limiting the number of threads processing nodes is important to keep the application from saturating the CPU with work, which causes slowdowns in the rest of the system. If the library is being used in a GUI application, this slowdown will affect that as well.

So it's important to limit the number of nodes that are processed at any one time.

For this to work smoothly it needs to take `NodeType::Graph` nodes into consideration as well. Each graph node can't be using all cores, the work queue used by an `Engine` needs to also be shared by any graph nodes inside it.

## Step 1
First step is to implement something that works, but not with all features. In this case that means having a work queue for a single graph, not working with nested graphs.

- [x] Write test for logic using dummy functions
- [x] Implement logic
- [x] Ability to change this setting during runtime
- [x] Default to the number of cores

## Step 2
Now the prioritisation system is relevant. The idea is to fill the work queue with prioritised nodes first, then fill the rest with regular requested nodes.

This helps make GUI applications more responsive as we can prioritise for instance the node that the user is editing right now, or all the nodes on the screen.

- [x] Write test for prioritisation system
- [x] Hook up the prioritisation system

## Step 3
Next step is to make all this work with nested graphs.

- [x] Implement logic

## Idea for tests
Have a heavy node with a high priority, and many light nodes with low priority. Calculate at most one node at a time. Continuously check if nodes are done, and if the heavy one is done first then it works.

**Comment by lukors** [comment]
This is done now.
